pkg/controller/cloudantv1/cloudantdatabase.go

```diff
@@ -0,0 +1,208 @@
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ibmc.IsResourceNotFound, err), errGetCloudantDatabaseFailed)
+	}
+
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	if err = ibmccdb.LateInitializeSpec(&cr.Spec.ForProvider, instance); err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrManagedUpdateFailed)
+	}
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		if err := c.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrManagedUpdateFailed)
+		}
+	}
+
+	cr.Status.AtProvider, err = ibmccdb.GenerateObservation(instance)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrGenObservation)
+	}
+
+	cr.Status.AtProvider.State = "active"
+
+	switch cr.Status.AtProvider.State {
+	case "active":
+		cr.Status.SetConditions(runtimev1alpha1.Available())
+	default:
+		cr.Status.SetConditions(runtimev1alpha1.Unavailable())
+	}
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	// have to ensure isuptodate always return true ?? just a note
+	upToDate, err := ibmccdb.IsUpToDate(&cr.Spec.ForProvider, instance, c.logger)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrCheckUpToDate)
+	}
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: nil,
+	}, nil
+}
+
+func (c *cloudantdatabaseExternal) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.CloudantDatabase)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotCloudantDatabase)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Creating())
+	resInstanceOptions := &cv1.PutDatabaseOptions{}
+	if err := ibmccdb.GenerateCreateCloudantDatabaseOptions(cr.Spec.ForProvider, resInstanceOptions); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateCloudantDatabaseOpts)
+	}
+
+	_, _, err := c.client.CloudantV1().PutDatabase(resInstanceOptions)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateCloudantDatabase)
+	}
+
+	meta.SetExternalName(cr, cr.Spec.ForProvider.Db)
+
+	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+}
+
+func (c *cloudantdatabaseExternal) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	return managed.ExternalUpdate{}, nil
+}
+
+func (c *cloudantdatabaseExternal) Delete(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*v1alpha1.CloudantDatabase)
+	if !ok {
+		return errors.New(errNotCloudantDatabase)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Deleting())
+
+	_, response, err := c.client.CloudantV1().DeleteDatabase(&cv1.DeleteDatabaseOptions{Db: reference.ToPtrValue(meta.GetExternalName(cr))})
+	// there might be a better way to stop the 202 problem ?? but this works for now
```

Comment:
This is good. My only suggestion is to use `http.StatusAccepted` from the http package (https://golang.org/src/net/http/status.go ) instead of directly the value 202